### DISCUSSION TITLE
Constant disconnections

### DISCUSSION THREAD

**Discussion by MondoBurrito** [community/community#11242] [Codespaces]
**EDIT: I loaded Codespaces on my computer at work. (Entirely different network) and did NOT receive any connection errors.
My home internet line is a 175mbps service via fiber.
My home network is a pfSense router with a Cisco layer 3 switch connecting all the devices in my house.
This network normally runs FLAWLESSLY, so perhaps it is showing a weakness with Codespaces. Again, any suggestions of what I can try would be appreciated.**

_ORIGINAL MESSAGE:_
So I am trying to use Codespace for CS50x. (web version)  I get CONSTANT disconnection errors.  It says "CANNOT CONNECT. PLEASE RELOAD THE WINDOW".  I have deleted the Codespace, and re-created it. I still get the errors. I have tried using desktop VSCode... I also get different errors, but results in the same problem.

Sometimes I can use Codespace for a few minutes before the error shows up.... but it usually starts popping up every 10 seconds. I then have to stop and re-start codespace. It is unusable.

I am using the latest version of Windows 10 with Chrome. I have no Chrome extensions running.

Here is the error:  [https://imgur.com/a/udHGIi1](https://imgur.com/a/udHGIi1)

I have talked to the sys admins at Harvard (CS50). They have no idea.
I checked my router logs to see if my pfSense router was blocking anything... I don't see any entries pop up when I start Codespace. I am at a loss here, and quite disappointed that I can not participate in CS50 because of this problem.

Any ideas at all?

Here is the debugging info:

`[
    {
        "id": 2910534,
        "name": "mondoburrito-code50-50547130-456j5wv52j7gr",
        "environment_id": "c1b95dad-bdc6-47c8-8c28-4c728895335a",
        "owner": {
            "login": "MondoBurrito",
            "id": 50547130,
            "node_id": "MDQ6VXNlcjUwNTQ3MTMw",
            "avatar_url": "https://avatars.githubusercontent.com/u/50547130?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/MondoBurrito",
            "html_url": "https://github.com/MondoBurrito",
            "followers_url": "https://api.github.com/users/MondoBurrito/followers",
            "following_url": "[https://api.github.com/users/MondoBurrito/following{/other_user}](https://api.github.com/users/MondoBurrito/following%7B/other_user%7D)",
            "gists_url": "[https://api.github.com/users/MondoBurrito/gists{/gist_id}](https://api.github.com/users/MondoBurrito/gists%7B/gist_id%7D)",
            "starred_url": "[https://api.github.com/users/MondoBurrito/starred{/owner}{/repo}](https://api.github.com/users/MondoBurrito/starred%7B/owner%7D%7B/repo%7D)",
            "subscriptions_url": "https://api.github.com/users/MondoBurrito/subscriptions",
            "organizations_url": "https://api.github.com/users/MondoBurrito/orgs",
            "repos_url": "https://api.github.com/users/MondoBurrito/repos",
            "events_url": "[https://api.github.com/users/MondoBurrito/events{/privacy}](https://api.github.com/users/MondoBurrito/events%7B/privacy%7D)",
            "received_events_url": "https://api.github.com/users/MondoBurrito/received_events",
            "type": "User",
            "site_admin": false
        },
        "billable_owner": {
            "login": "code50",
            "id": 89367332,
            "node_id": "MDEyOk9yZ2FuaXphdGlvbjg5MzY3MzMy",
            "avatar_url": "https://avatars.githubusercontent.com/u/89367332?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/code50",
            "html_url": "https://github.com/code50",
            "followers_url": "https://api.github.com/users/code50/followers",
            "following_url": "[https://api.github.com/users/code50/following{/other_user}](https://api.github.com/users/code50/following%7B/other_user%7D)",
            "gists_url": "[https://api.github.com/users/code50/gists{/gist_id}](https://api.github.com/users/code50/gists%7B/gist_id%7D)",
            "starred_url": "[https://api.github.com/users/code50/starred{/owner}{/repo}](https://api.github.com/users/code50/starred%7B/owner%7D%7B/repo%7D)",
            "subscriptions_url": "https://api.github.com/users/code50/subscriptions",
            "organizations_url": "https://api.github.com/users/code50/orgs",
            "repos_url": "https://api.github.com/users/code50/repos",
            "events_url": "[https://api.github.com/users/code50/events{/privacy}](https://api.github.com/users/code50/events%7B/privacy%7D)",
            "received_events_url": "https://api.github.com/users/code50/received_events",
            "type": "Organization",
            "site_admin": false
        },
        "repository": {
            "id": 456321512,
            "node_id": "R_kgDOGzLp6A",
            "name": "50547130",
            "full_name": "code50/50547130",
            "private": true,
            "owner": {
                "login": "code50",
                "id": 89367332,
                "node_id": "MDEyOk9yZ2FuaXphdGlvbjg5MzY3MzMy",
                "avatar_url": "https://avatars.githubusercontent.com/u/89367332?v=4",
                "gravatar_id": "",
                "url": "https://api.github.com/users/code50",
                "html_url": "https://github.com/code50",
                "followers_url": "https://api.github.com/users/code50/followers",
                "following_url": "[https://api.github.com/users/code50/following{/other_user}](https://api.github.com/users/code50/following%7B/other_user%7D)",
                "gists_url": "[https://api.github.com/users/code50/gists{/gist_id}](https://api.github.com/users/code50/gists%7B/gist_id%7D)",
                "starred_url": "[https://api.github.com/users/code50/starred{/owner}{/repo}](https://api.github.com/users/code50/starred%7B/owner%7D%7B/repo%7D)",
                "subscriptions_url": "https://api.github.com/users/code50/subscriptions",
                "organizations_url": "https://api.github.com/users/code50/orgs",
                "repos_url": "https://api.github.com/users/code50/repos",
                "events_url": "[https://api.github.com/users/code50/events{/privacy}](https://api.github.com/users/code50/events%7B/privacy%7D)",
                "received_events_url": "https://api.github.com/users/code50/received_events",
                "type": "Organization",
                "site_admin": false
            },
            "html_url": "https://github.com/code50/50547130",
            "description": null,
            "fork": false,
            "url": "https://api.github.com/repos/code50/50547130",
            "forks_url": "https://api.github.com/repos/code50/50547130/forks",
            "keys_url": "[https://api.github.com/repos/code50/50547130/keys{/key_id}](https://api.github.com/repos/code50/50547130/keys%7B/key_id%7D)",
            "collaborators_url": "[https://api.github.com/repos/code50/50547130/collaborators{/collaborator}](https://api.github.com/repos/code50/50547130/collaborators%7B/collaborator%7D)",
            "teams_url": "https://api.github.com/repos/code50/50547130/teams",
            "hooks_url": "https://api.github.com/repos/code50/50547130/hooks",
            "issue_events_url": "[https://api.github.com/repos/code50/50547130/issues/events{/number}](https://api.github.com/repos/code50/50547130/issues/events%7B/number%7D)",
            "events_url": "https://api.github.com/repos/code50/50547130/events",
            "assignees_url": "[https://api.github.com/repos/code50/50547130/assignees{/user}](https://api.github.com/repos/code50/50547130/assignees%7B/user%7D)",
            "branches_url": "[https://api.github.com/repos/code50/50547130/branches{/branch}](https://api.github.com/repos/code50/50547130/branches%7B/branch%7D)",
            "tags_url": "https://api.github.com/repos/code50/50547130/tags",
            "blobs_url": "[https://api.github.com/repos/code50/50547130/git/blobs{/sha}](https://api.github.com/repos/code50/50547130/git/blobs%7B/sha%7D)",
            "git_tags_url": "[https://api.github.com/repos/code50/50547130/git/tags{/sha}](https://api.github.com/repos/code50/50547130/git/tags%7B/sha%7D)",
            "git_refs_url": "[https://api.github.com/repos/code50/50547130/git/refs{/sha}](https://api.github.com/repos/code50/50547130/git/refs%7B/sha%7D)",
            "trees_url": "[https://api.github.com/repos/code50/50547130/git/trees{/sha}](https://api.github.com/repos/code50/50547130/git/trees%7B/sha%7D)",
            "statuses_url": "[https://api.github.com/repos/code50/50547130/statuses/{sha}](https://api.github.com/repos/code50/50547130/statuses/%7Bsha%7D)",
            "languages_url": "https://api.github.com/repos/code50/50547130/languages",
            "stargazers_url": "https://api.github.com/repos/code50/50547130/stargazers",
            "contributors_url": "https://api.github.com/repos/code50/50547130/contributors",
            "subscribers_url": "https://api.github.com/repos/code50/50547130/subscribers",
            "subscription_url": "https://api.github.com/repos/code50/50547130/subscription",
            "commits_url": "[https://api.github.com/repos/code50/50547130/commits{/sha}](https://api.github.com/repos/code50/50547130/commits%7B/sha%7D)",
            "git_commits_url": "[https://api.github.com/repos/code50/50547130/git/commits{/sha}](https://api.github.com/repos/code50/50547130/git/commits%7B/sha%7D)",
            "comments_url": "[https://api.github.com/repos/code50/50547130/comments{/number}](https://api.github.com/repos/code50/50547130/comments%7B/number%7D)",
            "issue_comment_url": "[https://api.github.com/repos/code50/50547130/issues/comments{/number}](https://api.github.com/repos/code50/50547130/issues/comments%7B/number%7D)",
            "contents_url": "[https://api.github.com/repos/code50/50547130/contents/{+path}](https://api.github.com/repos/code50/50547130/contents/%7B+path%7D)",
            "compare_url": "[https://api.github.com/repos/code50/50547130/compare/{base}...{head}](https://api.github.com/repos/code50/50547130/compare/%7Bbase%7D...%7Bhead%7D)",
            "merges_url": "https://api.github.com/repos/code50/50547130/merges",
            "archive_url": "[https://api.github.com/repos/code50/50547130/{archive_format}{/ref}](https://api.github.com/repos/code50/50547130/%7Barchive_format%7D%7B/ref%7D)",
            "downloads_url": "https://api.github.com/repos/code50/50547130/downloads",
            "issues_url": "[https://api.github.com/repos/code50/50547130/issues{/number}](https://api.github.com/repos/code50/50547130/issues%7B/number%7D)",
            "pulls_url": "[https://api.github.com/repos/code50/50547130/pulls{/number}](https://api.github.com/repos/code50/50547130/pulls%7B/number%7D)",
            "milestones_url": "[https://api.github.com/repos/code50/50547130/milestones{/number}](https://api.github.com/repos/code50/50547130/milestones%7B/number%7D)",
            "notifications_url": "[https://api.github.com/repos/code50/50547130/notifications{?since,all,participating}](https://api.github.com/repos/code50/50547130/notifications%7B?since,all,participating%7D)",
            "labels_url": "[https://api.github.com/repos/code50/50547130/labels{/name}](https://api.github.com/repos/code50/50547130/labels%7B/name%7D)",
            "releases_url": "[https://api.github.com/repos/code50/50547130/releases{/id}](https://api.github.com/repos/code50/50547130/releases%7B/id%7D)",
            "deployments_url": "https://api.github.com/repos/code50/50547130/deployments"
        },
        "machine": {
            "name": "basicLinux32gb",
            "display_name": "2 cores, 4 GB RAM, 32 GB storage",
            "operating_system": "linux",
            "storage_in_bytes": 34359738368,
            "memory_in_bytes": 4294967296,
            "cpus": 2,
            "prebuild_availability": null
        },
        "prebuild": null,
        "created_at": "2022-02-06T19:33:28-06:00",
        "updated_at": "2022-02-08T22:33:00-06:00",
        "last_used_at": "2022-02-08T22:28:07-06:00",
        "state": "Available",
        "url": "https://api.github.com/user/codespaces/mondoburrito-code50-50547130-456j5wv52j7gr",
        "git_status": {
            "ahead": 0,
            "behind": 0,
            "has_unpushed_changes": false,
            "has_uncommitted_changes": false,
            "ref": "main"
        },
        "location": "EastUs",
        "idle_timeout_minutes": 30,
        "web_url": "https://mondoburrito-code50-50547130-456j5wv52j7gr.github.dev/",
        "machines_url": "https://api.github.com/user/codespaces/mondoburrito-code50-50547130-456j5wv52j7gr/machines",
        "start_url": "https://api.github.com/user/codespaces/mondoburrito-code50-50547130-456j5wv52j7gr/start",
        "stop_url": "https://api.github.com/user/codespaces/mondoburrito-code50-50547130-456j5wv52j7gr/stop",
        "pulls_url": null,
        "recent_folders": [
            "/workspaces/50547130"
        ],
        "runtime_constraints": {
            "allowed_port_privacy_settings": [
                "private",
                "public",
                "org"
            ]
        }
    }
]`

**Comment by Reynald03** [comment]
I am also having the exact same problems as the poster above. Everything is working fine from week 1 to 3 and was able to submit all problems. Now I can not continue. Please can someone help. Thanks.

**Comment by bohblue2** [comment]
Same, Very fucked ux, Fuck codespace.
